packages/clay-tooltip/src/TooltipProvider.tsx

```diff
@@ -117,18 +133,49 @@ type TContentRenderer = (props: {
 	title: string;
 }) => React.ReactElement | React.ReactNode;
 
-const TooltipProvider: React.FunctionComponent<{
+interface IPropsBase {
+	/**
+	 * Flag to indicate if tooltip should automatically align based on the window
+	 */
 	autoAlign?: boolean;
-	children: React.ReactElement;
+
+	/**
+	 * Custom function for rendering the contents of the tooltip
+	 */
 	contentRenderer?: TContentRenderer;
+
+	/**
+	 * Delay in miliseconds before showing tooltip
+	 */
 	delay?: number;
-}> = ({
+}
+
+interface IPropsWithChildren extends IPropsBase {
+	children: React.ReactElement;
+	scope?: never;
+}
+
+interface IPropsWithScope extends IPropsBase {
+	children?: never;
+
+	/**
+	 * CSS selector to scope provider to. All titles within this scope will be
+	 * rendered in the tooltip. Titles outside of this scope will be styled
+	 * as with the default browser.
+	 */
+	scope: string;
```

Comment:
We can also accept the use of `Array<string>`, considering the case of DXP. What do you think?

Reply:
I don't think we accept an array in DXP, I got the logic from our delegate in DXP and you can see that [it expects a string](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/delegate/delegate.es.js#L30). You can use a comma to delaminate multiple selectors though (`.one-container, .another-container`).

Reply:
Oh ok, np. If I'm not mistaken I think that `querySelector` can accept the parameter as` Array<string>` or just `string`. But I'm not sure if it works for the `closest`. I looked at the DXP example, and I had seen an Array and not a literal template 😂 sorry.

Reply:
@matuzalemsteles no worries! I actually thought the same thing at first and I think you are right that both querySelector and closest automatically run `toString` on the argument and so it works with both string and array. I originally had an array for it and it was working fine so I was very confused. But I think it is safest to keep it as a string for now.